src/runtime/super.cpp

```diff
@@ -187,6 +187,36 @@ Box* superInit(Box* _self, Box* _type, Box* obj) {
     return None;
 }
 
+static PyObject* superDescrGet(PyObject* self, PyObject* obj, PyObject* type) {
+    BoxedSuper* su = static_cast<BoxedSuper*>(self);
+    BoxedSuper* newobj;
+
+    if (obj == NULL || obj == None || su->obj != NULL) {
+        /* Not binding to an object, or already bound */
+        Py_INCREF(self);
+        return self;
+    }
+    if (su->cls != super_cls) {
+        /* If su is an instance of a (strict) subclass of super,
+           call its type */
+        return runtimeCall(su->cls, ArgPassSpec(2, 0, false, false), su->type, obj, NULL, NULL, NULL);
+    } else {
+        /* Inline the common case */
+        BoxedClass* obj_type = supercheck(su->type, obj);
+        if (obj_type == NULL)
+            return NULL;
+        newobj = new (su->cls) BoxedSuper(NULL, NULL, NULL);
+        if (newobj == NULL)
+            return NULL;
```

Comment:
These should be `throwCapiException()`.  I usually prefer though to just leave the CPython code unmodified, ie have this be a capi-style function and stick it into tp_descr_get.